src/GenericCyclotomicFractions.jl

```diff
@@ -23,34 +23,12 @@ function shrink(a::GenericCycloFrac)  # TODO Move this to the constructor of Gen
   return GenericCycloFrac(new_numerator, new_denominator, a.exceptions; simplify=false)
 end
 
-function show(io::IO, x::GenericCycloFrac)
-  io = pretty(io)
-  if isone(x.denominator)
-    print(io, "$(x.numerator)")
-  else
-    if isone(length(x.numerator.f))
-      print(io, "$(x.numerator)//")
-    else
-      print(io, "($(x.numerator))//")
-    end
-    if isone(length(x.denominator.f))
-      argument, modulus = collect(x.denominator.f)[1]
-      if iszero(argument) && (is_monomial(modulus) || is_constant(modulus))
-        print(io, "$(x.denominator)")
-      else
-        print(io, "($(x.denominator))")
-      end
-    else
-      print(io, "($(x.denominator))")
-    end
-  end
-  if is_restriction(x.exceptions)
-    print(io, "\nWith exceptions:\n", Indent())
-    print(io, x.exceptions)
-    print(io, Dedent())
-  end
+function expressify(x::GenericCycloFrac; context = nothing)
+  return Expr(:call, ://, expressify(x.numerator, context = context), expressify(x.denominator, context = context))
```

Comment:
To avoid the `1//1` issue, you could just do this:
```suggestion
  n = expressify(x.numerator; context)
  isone(x.denominator) && return n
  d = expressify(x.denominator; context)
  return Expr(:call, ://, n, d)
```
at least if testing `isone(x.denominator)` is reasonable here... If not one can always fallback to testing purely syntactically:
```suggestion
  n = expressify(x.numerator; context)
  d = expressify(x.denominator; context)
  is_syntactic_one(d) && return n
  return Expr(:call, ://, n, d)
```

Reply:
The first version is sufficient here.